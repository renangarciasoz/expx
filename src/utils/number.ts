import { ONE_BILLION } from "constants/numbers.constants";

export const formatValue = (value: number, minimumFractionDigits = 2) =>
  Intl.NumberFormat("en", {
    style: "currency",
    currency: "USD",
    minimumFractionDigits,
  }).format(value);

export const formatBillionValue = (
  value: string | number,
  minimumFractionDigits = 3
) => {
  return `${formatValue(Number(value) / ONE_BILLION, minimumFractionDigits)}b`;
};
