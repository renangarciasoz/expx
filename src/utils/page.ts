export const blockPageScroll = () => {
  const body = document?.querySelector("body");
  body!.style.height = "100vh";
  body!.style.overflow = "hidden";
};

export const unblockPageScroll = () => {
  const body = document?.querySelector("body");
  body!.style.height = "auto";
  body!.style.overflow = "auto";
};
