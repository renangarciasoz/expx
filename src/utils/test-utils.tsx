import { render, RenderOptions } from "@testing-library/react"; //jest-ignore
import { i18n } from "configs/i18next-tests.config";
import { ThemeProvider } from "contexts/theme.ctx";
import React, { ReactElement } from "react";
import { I18nextProvider } from "react-i18next";

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ThemeProvider>
      <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
    </ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
