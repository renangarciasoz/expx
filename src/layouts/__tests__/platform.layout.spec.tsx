/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "utils/test-utils";
import { PlatformLayout } from "../platform.layout";

describe("platform layout", () => {
  it("renders correctly", () => {
    const { asFragment, getByText } = render(
      <PlatformLayout>
        <div>Hello World</div>
      </PlatformLayout>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByText("Hello World")).toBeInTheDocument();
  });
});
