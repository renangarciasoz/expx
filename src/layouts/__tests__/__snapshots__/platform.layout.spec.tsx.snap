// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`platform layout renders correctly 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-10d26i6-MuiPaper-root-MuiAppBar-root"
    data-testid="header"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1prgphj-MuiToolbar-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 css-1ht2c9-MuiTypography-root"
        data-testid="logo"
      >
        ExPx
      </h1>
      <div
        class="MuiBox-root css-t508yj"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="change-header-position-btn"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 10V8h-2v4H8l4 4 4-4h-3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-1j9bsaa"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-xn5ipf-MuiButtonBase-root-MuiButton-root"
          data-testid="login-btn"
          tabindex="0"
          type="button"
        >
          Login
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <main
    class="MuiContainer-root MuiContainer-maxWidthLg css-1i8s1pv-MuiContainer-root"
  >
    <div>
      Hello World
    </div>
  </main>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
  <footer
    class="MuiBox-root css-6fytqd"
  >
    <p
      class="MuiTypography-root MuiTypography-caption css-1bhdj8e-MuiTypography-root"
    >
      All rights reserved to ExPx
    </p>
    <p
      class="MuiTypography-root MuiTypography-caption css-1bhdj8e-MuiTypography-root"
    >
      Made and Designed by 
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1tf71kh-MuiTypography-root-MuiLink-root"
        href="https://renangarcia.com"
      >
        Renan Garcia
      </a>
    </p>
  </footer>
</DocumentFragment>
`;
